src/csm_share/streams/shr_stream_mod.F90

```diff
@@ -642,6 +656,78 @@ subroutine shr_stream_init(strm,infoFile,yearFirst,yearLast,yearAlign,taxMode,rc
 
     close(nUnit)
 
+#ifdef USE_PDAF
+    !-----------------------------------------------------------------------------
+    ! Yorck adaptations: read in time resolution of forcings and caseId, numEns
+    !-----------------------------------------------------------------------------
+
+    !--- find start tag ---
+    open(nUnit,file=infoFile,STATUS='OLD',FORM='FORMATTED',ACTION='READ')
+    startTag =  "<fieldInfo>"
+    endTag = "</fieldInfo>"
+    call shr_stream_readUpToTag(nUnit,startTag,rc=rCode2)
+    if (rCode2 /= 0)then
+       rCode = rCode2
+       goto 999
```

Comment:
[GOTO considered harmful](https://stackoverflow.com/questions/46586/goto-still-considered-harmful) 😜

Anyway the old code originally has this so I'll ignore it for now

For context here's the `goto` target:

https://github.com/HPSCTerrSys/eCLM/blob/7853ec4db711d9094df76ba642cb074fde07f5a6/src/csm_share/streams/shr_stream_mod.F90#L659-L665

Reply:
I agree about the `GOTO` :-)

I would leave it as is for now.

Remark: This may point to an underlying issue of `shr_stream_mod.F90` / streams and how they are differently implemented in future CLMs (CTSM)...